### PR TITLE
Feature/streak new container

### PR DESCRIPTION
**Pull request by rostikjoystick**
**YouTrack task**: [#APPS-2344](https://vyahhi.myjetbrains.com/youtrack/issue/APPS-2344)

**Description List**:
* Transferred streak dialog logic to new lesson container

### CODE REVIEW

File: app/src/main/java/org/stepik/android/domain/streak/interactor/StreakInteractor.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    private val sharedPreferenceHelper: SharedPreferenceHelper,
+    private val streakNotificationDelegate: StreakNotificationDelegate
+) {
+
+    fun needShowStreakDialog(): Boolean =
+        isStreakNotificationEnabled() &&
+        canShowStreakDialog() &&
+        isAuthResponseFromStore()
+
+    fun onNeedShowStreak(): Observable<Int> =
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/domain/streak/interactor/StreakInteractor.kt]: Тут можно что-то такое написать, чтобы обойтись без RxOptional

```kt
fun onNeedShowStreak(): Maybe<Int> =
        Maybe
            .fromCallable {
                sharedPreferenceHelper.profile?.id
            }
            .flatMapSingleElement { userActivityRepository.getUserActivities(it) }
            .flatMap { it.firstOrNull()?.pins.toMaybe() }
            .map { StepikUtil.getCurrentStreak(it) }
```

File: app/src/main/java/org/stepik/android/view/lesson/ui/activity/LessonActivity.kt [hunk 2]
```diff
@@ -311,6 +371,17 @@ class LessonActivity : FragmentActivityBase(), LessonView, NextMoveable, RateApp
...
 
+    override fun onTimeIntervalPicked(data: Intent) {
+        val intervalCode = data.getIntExtra(TimeIntervalPickerDialogFragment.RESULT_INTERVAL_CODE_KEY, TimeIntervalUtil.defaultTimeCode)
+        lessonPresenter.setStreakTime(intervalCode)
+        analytic.reportEvent(Analytic.Streak.CHOOSE_INTERVAL, intervalCode.toString())
+        Snackbar.make(lessonPager, R.string.streak_notification_enabled_successfully, Snackbar.LENGTH_LONG)
+                .setTextColor(ColorUtil.getColorArgb(R.color.white, baseContext))
+                .show()
+
+        // TODO StepAttemptFragment handled a cancelled state by showing a snackbar saying that the notifications were not enabled
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/view/lesson/ui/activity/LessonActivity.kt]: We can add another method to `TimeIntervalPickerDialogFragment.Callback` to handle it

Comment (eadm) [reply app/src/main/java/org/stepik/android/view/lesson/ui/activity/LessonActivity.kt]: Also it strange that `TimeIntervalPickerDialogFragment` uses Callback and onActivityResult logic at the same time

Comment (eadm) [reply app/src/main/java/org/stepik/android/view/lesson/ui/activity/LessonActivity.kt]: It seems that current callback logic is broken and doesn't handle configuration changes. It should be fixed or removed and only activity result used.

Comment (rostikjoystick) [reply app/src/main/java/org/stepik/android/view/lesson/ui/activity/LessonActivity.kt]: I have decided to removed the onActivityResult logic. The callback logic now handles configuration changes, so it should work fine

File: app/src/main/java/org/stepic/droid/ui/fragments/ProfileFragment.kt [hunk 3]
```diff
@@ -144,7 +142,7 @@ class   ProfileFragment : FragmentBase(),
             analytic.reportEvent(Analytic.Interaction.CLICK_CHOOSE_NOTIFICATION_INTERVAL)
             val dialogFragment = TimeIntervalPickerDialogFragment.newInstance()
             if (!dialogFragment.isAdded) {
-                dialogFragment.setTargetFragment(this@ProfileFragment, NOTIFICATION_INTERVAL_REQUEST_CODE)
+                dialogFragment.setTargetFragment(this@ProfileFragment, 0)
```
Comment (eadm) [inline app/src/main/java/org/stepic/droid/ui/fragments/ProfileFragment.kt]: We have `INTERVAL_CODE_KEY` but not using it

File: app/src/main/java/org/stepic/droid/ui/fragments/StepAttemptFragment.java [hunk 4]
```diff
@@ -619,24 +620,25 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
-                                        Snackbar.LENGTH_LONG),
-                                ColorUtil.INSTANCE.getColorArgb(R.color.white,
-                                        getContext()))
-                        .show();
-            } else if (resultCode == Activity.RESULT_CANCELED) {
-                getAnalytic().reportEvent(Analytic.Streak.CHOOSE_INTERVAL_CANCELED);
-                messageOnNotEnablingNotification();
-            }
         }
+//        else if (requestCode == NOTIFICATION_TIME_REQUEST_CODE) {
```
Comment (eadm) [inline app/src/main/java/org/stepic/droid/ui/fragments/StepAttemptFragment.java]: Commented code should be removed if it not needed

**Pull request merged**